### PR TITLE
add interface to default interactors

### PR DESCRIPTION
**Pull request by jfilowk**
Hello team,

Created the interfaces for our **Future** default interactors.
Executor is null in the interface because we want to support constructor and invoke declaration.

Also, I updated the sample to see the changes in the DI. Tomorrow, I would like to do the same in PeerWell and start testing.

Let's discuss these changes here in the PR or if you create an async chat in Asana, what you prefer.

### CODE REVIEW

**Comment by joselufo** [comment]
It's perfect. Start testing it and if everything is right after refactoring PW we can merge to master. 

BTW: `Interactors.kt` class is starting to be big, maybe we could split it in 3 files. 

- `GetInteractor.kt`
- `PutInteractor.kt`
- `DeleteInteractor.kt`

**Review by franmontiel — APPROVED** [review]

**Review by joselufo — APPROVED** [review]
I'm approving it but as discussed, NOT merge it to master till is validated in PW.

**Comment by newboadki** [comment]
This change won't keep backward-compatibility, will it? Since we are renaming the interactors, it will force client applications to get updated. 

If we named the protocols differently, client applications won't require changes to make client apps compile after the upgrade. 

Any thoughts on rethinking the convention? @joselufo , @jfilowk , @franmontiel

**Comment by joselufo** [comment]
Good point Borja but I personally think we can't give retro compatible because if not we will end up with a naming convention which is weird.

@newboadki maybe you could propose a retro compatible naming to be and we can discuss it.

**Comment by newboadki** [comment]
@joselufo, yes, the names we might find won't be ideal, since that name is in use: GetInteractor. One option could be to suffix Interface to the name. GetInteractorInterface. 

That's my two cents. If this was already considered and there's consensus on that client applications will need to rename all the references, then let's move on.

**Comment by joselufo** [comment]
I'm still thinking not giving retro-compatibility is the best thing here. I also don't think it's too much work as it could be solved with a "Find and Replace".

What do you think @jfilowk ?

**Comment by jfilowk** [comment]
Fran and I got this conversation about if we only change the name of the interfaces, everything will keep working without any refactoring. The point is that we want to be aware of it and if we change the default implementation, the change will be propagated and we have consistency.

My experience doing the PeerWell refactor is:
If you are using extensions (toXXXInteractor) everything will remain the same, at least in Kotlin implementation.
Find and replace is enough and doesn't require any other change, at least in Kotlin implementation.

**Comment by joselufo** [comment]
Ok, so I would say that we can keep it as it is.

**Comment by jfilowk** [comment]
I'm going to merge it and create the 1.0.9 version.

**Pull request merged**
